examples/serialization/avro/avro-cli.py

```diff
@@ -83,75 +75,90 @@ def on_delivery(err, msg):
-              msg.topic(), msg.partition(), msg.offset(), msg.value()))
+              msg.topic(), msg.partition(), msg.offset(), UserRecord(msg.value())))
         return 1
 
 
-def produce(args):
+def produce(topic, conf):
     from confluent_kafka.avro import AvroProducer
 
-    record = User()
-    topic = args.topic
-    conf = args.producer_conf()
-
-    producer = AvroProducer(conf, default_value_schema=record.schema())
+    producer = AvroProducer(conf, default_value_schema=UserRecord.schema())
 
     print("producing user records to topic {}. ^c to exit.".format(topic))
-
     while True:
+        # Instantiate new UserRecord, populate fields, produce record, execute callbacks.
         try:
+            record = UserRecord()
             record.prompt()
             producer.produce(topic=topic, partition=0, value=record, callback=on_delivery)
-            producer.poll(0.1)
+            producer.poll(0)
         except KeyboardInterrupt:
             break
         except ValueError:
-            print("Invalid input,  discarding record...")
+            print("Invalid input, discarding record...")
             continue
-        record = User()
 
     print("\nFlushing records...")
     producer.flush()
 
 
-def consume(args):
+def consume(topic, conf):
     from confluent_kafka.avro import AvroConsumer
     from confluent_kafka import KafkaError
     from confluent_kafka.avro.serializer import SerializerError
 
-    topic = args.topic
-    conf = args.consumer_conf()
-    print("consuming user records from topic {}".format(topic))
+    print("consuming user records from topic {} with group {}".format(topic, conf["group.id"]))
 
-    c = AvroConsumer(conf)
-    c.subscribe([topic])
+    # Configure avro schema when the reader schema associated with our UserRecord
+    c = AvroConsumer(conf, reader_value_schema=UserRecord.schema())
+    c.assign([TopicPartition(topic, 0, 0)])
 
+    # Consume until EOF; contine runnning when faced with errors
     while True:
         try:
             msg = c.poll(1)
-
+        # Report malformed record, discard results, continue polling
         except SerializerError as e:
             print("Message deserialization failed {}".format(e))
-            break
+            continue
 
+        # There were no messages on the queue, continue polling
         if msg is None:
             continue
 
+        # Exit if we have reached the end of the topic,
+        # otherwise report any errors and continue polling
         if msg.error():
             if msg.error().code() == KafkaError._PARTITION_EOF:
                 break
             else:
                 print(msg.error())
-                break
+                continue
 
-        record = User(msg.value())
-        print("\nusername: {}\n\tfavorite_number:{}\n\tfavorite_color:{}".format(
-              record["name"], record["favorite_number"], record["favorite_color"]))
+        print(UserRecord(**msg.value()))
```

Comment:
For clarity I suggest instantiating the UserRecord first, and then printing it.
Also, it would look better if we could get rid of the `**`

Reply:
And I would still look to see individual UserRecord fields being printed to show case that this is actually a usable object after deserialization, relying on __str__ doesn't really say much.